### DISCUSSION TITLE
How to handle cookies?

### DISCUSSION THREAD

**Discussion by nbigaouette** [DioxusLabs/dioxus#2473] [Q&A]
Hi all! I have written a simple backend (in axum) and now trying to write a frontend for it. There's a simple API but it requires a cookie to be set to identify clients. From what I understand the webassembly-compiled dioxus app does not have access to the browser cookie; backend queries happen through an http client (ie reqwest) which is independent of the browser.

That means that the app needs to perform its own reqwest requests and handle the cookie itself. Can I share a reqwest::Client across the app (so that cookies are kept and reused)? I don't know if this is possible, when I tried rustc complained that reqwest's `ClientBuilder` didn't have the `cookie_store()` method, even though I enabled the `cookies` cargo feature.

In any case, what is the proper way to make backend requests and saving/providing cookies?

Thanks!

**Comment by hmziqrs** [comment]
I spend a good amount of time figuring out cookies with reqwest, but an issue for this has been open for quite some time on reqwest github repo with no sign of progress. I wrote a custom solution using gloo-net http and wasm-cookies, which works quite nicely.

https://crates.io/crates/wasm-cookies
https://docs.rs/gloo-net/latest/gloo_net/http/index.html
